File: ftgo-kitchen-service/src/main/java/me/jangjunha/ftgo/kitchen_service/grpc/ServerConfiguration.java
package me.jangjunha.ftgo.kitchen_service.grpc;

import io.grpc.ManagedChannelBuilder;
import me.jangjunha.ftgo.kitchen_service.service.KitchenService;
import me.jangjunha.ftgo.order_service.api.OrderServiceGrpc;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties(Destinations.class)
public class ServerConfiguration {

    @Value("${grpc.server.port}")
    private int grpcServerPort;

    @Bean
    public GrpcServer grpcServer(KitchenService kitchenService, OrderServiceGrpc.OrderServiceBlockingStub orderService) {
        return new GrpcServer(grpcServerPort, kitchenService, orderService);
    }

    @Bean
    public OrderServiceGrpc.OrderServiceBlockingStub orderService(Destinations destinations) {
        return OrderServiceGrpc.newBlockingStub(ManagedChannelBuilder.forTarget(destinations.getOrderServiceUrl()).usePlaintext().build());
    }
}

File: ftgo-delivery-service/src/main/kotlin/me/jangjunha/ftgo/delivery_service/grpc/ServerConfiguration.kt
package me.jangjunha.ftgo.delivery_service.grpc

import me.jangjunha.ftgo.delivery_service.domain.DeliveryService
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ServerConfiguration {

    @Value("\${grpc.server.port}")
    private val grpcServerPort = 8108

    @Bean
    fun grpcServer(deliveryService: DeliveryService): GrpcServer {
        return GrpcServer(grpcServerPort, deliveryService)
    }
}
